app/src/main/java/com/hedvig/app/feature/embark/passages/multiaction/MultiActionAdapter.kt

```diff
@@ -0,0 +1,66 @@
+package com.hedvig.app.feature.embark.passages.multiaction
+
+import android.view.View
+import android.view.ViewGroup
+import android.widget.FrameLayout
+import android.widget.TextView
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import com.hedvig.app.R
+import com.hedvig.app.util.GenericDiffUtilItemCallback
+import com.hedvig.app.util.extensions.inflate
+
+class MultiActionAdapter(private val clickListener: ClickListener) : ListAdapter<MultiAction, MultiActionViewHolder>(GenericDiffUtilItemCallback()) {
+
+    interface ClickListener {
```

Comment:
A function would be ideal, but cant use function references with parcelize. I prefer using an interface in this case since it decouples the viewmodel from the adapter.

Reply:
How about something like:

```kotlin
class MultiActionAdapter(private val onComponentClick: (Long) -> Unit, private val onComponentRemove: (Long) -> Unit
```

which we could then call like this:

```kotlin
someRecyclerView.adapter = MultiActionAdapter(viewModel::onComponentClick, viewModel::onComponentRemove)
```